bpfd/src/bpf.rs

```diff
@@ -48,13 +48,138 @@ const MAPS_MODE: u32 = 0o0660;
 
 pub(crate) struct BpfManager {
     config: Config,
-    dispatchers: HashMap<DispatcherId, Dispatcher>,
-    programs: HashMap<Uuid, Program>,
+    dispatchers: BpfDispatchers,
+    programs: BpfPrograms,
     maps: HashMap<Uuid, BpfMap>,
     commands: Receiver<Command>,
     image_manager: Sender<ImageManagerCommand>,
 }
 
+pub(crate) struct BpfPrograms {
```

Comment:
Note: We *could* make this type `UuidMap` and it would become:

```rust
pub(crate) struct UuidMap<T> {
   inner: HashMap<Uuid, T>
}
```

You could then use this same struct for `BpfManager::maps`.
So you'd have `UuidMap<Dispatcher>` and `UuidMap<BpfMap>`...

Reply:
Right but I think I'd rather have a dedicated Type for Maps especially as we enter the next stage of map development.

Specifically I prototyped removing the `BpfMap` type completely but ran into issues because now Maps have slightly different lifetimes then Programs (i.e a Map can hang around if a it's owner program was deleted but other programs are still using it) 

Therefore I think it will make the most sense to move all of the map functionality we have today into traits on a ProgramMap type (no more passing around Owned UUIDs /IDS everwhere YAY!)